runners/core-java/src/test/java/org/apache/beam/runners/core/SimpleDoFnRunnerTest.java

```diff
@@ -472,6 +659,40 @@ public Duration getAllowedTimestampSkew() {
     }
   }
 
+  /**
+   * A {@link DoFn} that creates/sets a timer with an output timestamp equal to the input timestamp
+   * minus the input element's value. Keys are ignored but required for timers.
+   */
+  private static class TimerSkewingDoFn extends DoFn<KV<String, Duration>, Duration> {
+    static final String TIMER_ID = "testTimerFamily";
+    private final Duration allowedSkew;
+
+    @TimerFamily(TIMER_ID)
+    private final TimerSpec timers = TimerSpecs.timerMap(TimeDomain.PROCESSING_TIME);
```

Comment:
Can you not use `TimerMap` as it limits the number of runners this can run on and use individual timers?

Reply:
The elements need to share a timer then but I suppose the same logic will hit whenever we set the output timestamp. I removed the output of elements though since this would be harder. Wasn't really needed anyway.